oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,483 @@
+use the ADL_ standard `Experience API (xAPI)`_ as the communication protocol between
+the Open edX LMS and registered adaptive engines. We have chosen xAPI since it has
+`widespread industry support and usage`_ for tracking and analyzing learning activity.
+Please see `xAPI Architecture Overview`_ for further information on xAPI.
+
+.. _ADL: https://adlnet.gov/
+.. _Experience API (xAPI): https://www.adlnet.gov/research/performance-tracking-analysis/experience-api/
+.. _widespread industry support and usage: https://xapi.com/adopters/
+.. _xAPI Architecture Overview: https://www.adlnet.gov/research/performance-tracking-analysis/experience-api/xapi-architecture-overview/
+
+Learning Record Store (LRS) - Future
+====================================
+
+xAPI includes specification for a Learning Record Store (LRS), which encapsulates a
+data store with APIs for storing and accessing xAPI data. While an LRS could be used
+for many different purposes, it can be used to train adaptive engines as shown in the
+diagram below.
+
+.. image:: oep-0026/adaptive_learning_lrs_basic.png
+   :alt: The diagram above is enhanced with a new LRS component that receives events
+    from the Open edX "Eventing" component and is accessed by adaptive engines for
+    training purposes, using xAPI for both transactions.
+
+In the short term, however, we will not implement our own LRS since the initial integration
+efforts do not require this (as shown in the first diagram above).
+
+Statement API
+=============
+
+xAPI specifies 4 standard REST-ful JSON payload APIs, of which only the `Statement API`_
+is needed for our purposes as that API is sufficient for tracking learning activity.
+
+The Statement API has essentially the following parts: `xAPI Actor`_, `xAPI Verb`_,
+`xAPI Object`_, `xAPI Context`_, and `xAPI Result`_.
+
+.. _Statement API: https://xapi.com/statements-101/
+
+xAPI Actor
+~~~~~~~~~~
+Although the **Actor** field can be either an *Agent* or a *Group*, we will primarily
+support only the *Agent* type, which is used for individuals performing an activity
+(`xAPI Verb`_ on an `xAPI Object`_).
+
+An Actor can be identified using `Friend of a Friend (FOAF)`_ vocabulary with either:
+(1) `email address`_, (2) `hash of email address`_, (3) `OpenID URI`_, or (4) `account`_
+with a *homepage*-scoped identifier.  One of these is sent along with the Actor's "name".
+To be mindful of learner privacy, we will initially take a conservative approach
+and only send #3, with an Open edX anonymized unique identifier of the learner
+(`Anonymized User ID`_).
+
+In the future, if certain external systems require `Personally Identifiable Information
+(PII)`_, like the learner's email address or name, then those may be conditionally sent
+with appropriate permissions. Adaptive engines, however, do not need PII.
+
+Initially, we will exclude the "name" field. However, if we find that xAPI JSON parsers
```

Comment:
I think we don't need name - we took it off, but let us know.

Reply:
Great.